src/extensions/external-integrations.ts

```diff
@@ -0,0 +1,36 @@
+import { PostHog } from '../posthog-core'
+import { assignableWindow, PostHogExtensionKind } from '../utils/globals'
+import { createLogger } from '../utils/logger'
+
+const logger = createLogger('[ExternalIntegrations]')
+
+export class ExternalIntegrations {
+    constructor(private readonly _instance: PostHog) {}
+
+    private _loadScript(name: PostHogExtensionKind, cb: () => void): void {
+        assignableWindow.__PosthogExtensions__?.loadExternalDependency?.(this._instance, name, (err) => {
+            if (err) {
+                return logger.error('failed to load script', err)
+            }
+            cb()
+        })
+    }
+    public startIfEnabledOrStop() {
+        if (
+            this._instance.config.integrations?.intercom &&
+            !assignableWindow.__PosthogExtensions__?.integrations?.intercom
+        ) {
+            this._loadScript('intercom-integration', () => {
+                assignableWindow.__PosthogExtensions__?.integrations?.intercom?.start(this._instance)
+            })
+        }
+        if (
+            this._instance.config.integrations?.crispChat &&
+            !assignableWindow.__PosthogExtensions__?.integrations?.crispChat
+        ) {
+            this._loadScript('crisp-chat-integration', () => {
+                assignableWindow.__PosthogExtensions__?.integrations?.crispChat?.start(this._instance)
+            })
+        }
```

Comment:
might be possible to squash this duplication, particularly if we add more of these

Reply:
@pauldambra just commited 898c756a7b43823971fcfc58071838c1c61bc245 wdyt?